spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/cache/CacheAutoConfigurationTests.java

```diff
@@ -97,6 +99,16 @@
 
 	private AnnotationConfigApplicationContext context;
 
+	@BeforeClass
+	public static void init() {
+		System.setProperty("hazelcast.jcache.provider.type", "server");
+	}
+
+	@AfterClass
+	public static void close() {
+		System.clearProperty("hazelcast.jcache.provider.type");
+	}
+
```

Comment:
If that's the case, that's what the test should assert then. And you should have a test that filters out the client (see that JUnit rule) to restore what happens when the client jar isn't on the classpath.